action/metadata-support.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/sethvargo/go-githubactions"
+	"golang.org/x/oauth2"
+)
+
+// getArtifacts queries github for artifact list, returning a 2 dimensional slice of
+// artifact names paired with their variant type.
+// repo expects name of the product as it appears in its artifacts
+// as completely as possible, for example, "vault" for oss vault and "vault-enterprise"
+// for enterprise vault.
+func getArtifacts(org string, repo string, workflowRunID int64) [][]string {
+
+	// Auth to github API.
+	token := os.Getenv("CRT_GITHUB_TOKEN")
+	if token == "" {
+		githubactions.Errorf("missing env CRT_GITHUB_TOKEN")
+	}
+
+	ctx := context.Background()
```

Comment:
Question: What does `context.Background()` do here? (mostly because I'm a newb and genuinely want to understand 😅)

Reply:
It creates an empty context - I don't implement anything outside of having one created here but it is used when interacting with APIs if you discover timeout issues or with goroutines.